rosbag2_py/src/rosbag2_py/format_bag_metadata.cpp

```diff
@@ -139,13 +142,29 @@ void format_topics_with_type(
       info_stream << std::endl;
     };
 
+  std::vector<size_t> sorted_idx(topics.size());
+  std::iota(sorted_idx.begin(), sorted_idx.end(), 0);
+  std::sort(
+    sorted_idx.begin(),
+    sorted_idx.end(),
+    [&topics, sort_method](size_t i1, size_t i2) {
+      if (sort_method == rosbag2_py::SortingMethod::TYPE) {
+        return topics[i1].topic_metadata.type < topics[i2].topic_metadata.type;
+      }
+      if (sort_method == rosbag2_py::SortingMethod::COUNT) {
+        return topics[i1].message_count < topics[i2].message_count;
+      }
+      return topics[i1].topic_metadata.name < topics[i2].topic_metadata.name;
```

Comment:
Please use switch-case instead of if-else with returns.
In case of `default:` "unknown value" in switch case, please throw std::runtime_error("switch is not exhaustive");